src/datanode/src/instance.rs

```diff
@@ -273,7 +288,21 @@ pub(crate) async fn new_s3_object_store(store_config: &ObjectStoreConfig) -> Res
         config: store_config.clone(),
     })?;
 
-    Ok(ObjectStore::new(accessor))
+    let mut object_store = ObjectStore::new(accessor);
+    if let Some(path) = s3_config.cache_path.as_ref() {
+        let accessor =
```

Comment:
Can we extract this code into a function `create_object_store_cache(object_store, path, capacity)`, and reuse it? Looks like it emerges twice.